api/src/main/java/org/datacleaner/api/AnalyzerResultFutureImpl.java

```diff
@@ -165,9 +167,17 @@ public boolean isReady() {
     public R get() {
         if (_countDownLatch != null) {
             try {
-                _countDownLatch.await();
+                boolean finished = false;
+                int iteration = 0;
+                while (!finished) {
+                    iteration++;
+                    finished = _countDownLatch.await(2, TimeUnit.MINUTES);
+                    if (!finished) {
+                        logger.info("Awaited completion for " + (iteration * 2) + " minutes...");
```

Comment:
Let's add a constant for the minute count. We may want to adjust it at some point, and it's easy to miss two values that must be in sync.